src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -65,15 +65,42 @@ public function add(array $parameters = [])
     }
 
     /**
-     * Returns a parameter by name.
+     * Returns a string parameter by name.
      *
-     * @param mixed $default The default value if the parameter key does not exist
+     * @param string|null $default The default value if the parameter key does not exist
      *
-     * @return mixed
+     * @return string|null
      */
     public function get(string $key, $default = null)
     {
-        return \array_key_exists($key, $this->parameters) ? $this->parameters[$key] : $default;
+        if (null !== $default && !\is_string($default)) {
+            @trigger_error(sprintf('Passing a non-string value as 2nd argument to "%s::%s()" is deprecated since Symfony 5.1, pass a string or null instead.', static::class, __METHOD__), E_USER_DEPRECATED);
+        }
+
+        $value = \array_key_exists($key, $this->parameters) ? $this->parameters[$key] : $default;
+
+        if (null !== $value && $default !== $value && !\is_string($value)) {
+            @trigger_error(sprintf('Retriving a non-string value from "%s::%s()" is deprecated since Symfony 5.1, use "%s::getAll()" instead.', static::class, __METHOD__, static::class), E_USER_DEPRECATED);
```

Comment:
That's not a viable change, `Request::$attributes` is an instance of `ParameterBag` and is often used to store non-string values (for example, objects retrieved by param converters).

Reply:
Hmm :thinking: totally forget about `Request::$attributes`, should this be moved to a subclass for `Request::$request`, `Request::$cookies`, `Request::$query` ? the point of this is to make people aware that there's no guarantee `$request->request->get('something')` will return a string. client have control of the return type being a string or an array.